opentracing_instrumentation/client_hooks/_patcher.py

```diff
@@ -0,0 +1,26 @@
+class Patcher(object):
+
+    patches_installed = False
```

Comment:
should this be a private var, i.e. `_patches_installed`?

Reply:
I don't think so since an ability to check whether patches are installed could be useful for some users.
As a programmer, I don't really like to see a warning in my IDE when I have to do something like
```python
from opentracing_instrumentation.client_hooks.requests import patcher

if pather._patches_installed:  # there will be a warning
    pass  # some logic here
```

Reply:
I think if you want to support this kind of check, you can expose a read-only `@property` with a public name. This field is a class-level field, so `pather.patches_installed` syntax on an instance is misleading anyway (rather than `Patcher.patches_installed`).

Reply:
> This field is a class-level field, so pather.patches_installed syntax on an instance is misleading anyway (rather than Patcher.patches_installed).

It's not true since `patches_installed` attribute of `Patcher` class was used as a default value only, just like [in that case](https://github.com/uber-common/opentracing-python-instrumentation/pull/66#discussion_r244413173).

Reply:
I remember, I just don't like that pattern of mixing class and instance level fields. It's confusing and not obvious, and in 6 months I won't remember those arguments (I also don't agree with them as you're advocating for a subclass approach while the RequestPatcher supports the same via simple composition). But I want to merge this PR since you've already put a lot of work into it.